beacon-chain/blockchain/service.go

```diff
@@ -340,7 +341,29 @@ func (c *ChainService) blockProcessing() {
 				continue
 			}
 			if !parentExists {
-				log.Debugf("Block points to nil parent", err)
+				log.Debugf("Block points to nil parent: %v", err)
+				continue
+			}
+			parent, err := c.chain.getBlock(block.ParentHash())
+			if err != nil {
+				log.Debugf("Could not get parent block: %v", err)
+				continue
+			}
+
+			aState := c.chain.ActiveState()
+			cState := c.chain.CrystallizedState()
+			isTransition := cState.IsCycleTransition(c.currentSlot - 1)
+
+			if isTransition {
+				cState, err = cState.NewStateRecalculations(aState, block)
+				if err != nil {
+					log.Errorf("Initialize new cycle transition failed: %v", err)
+					continue
+				}
+			}
+
+			if valid := block.IsValid(aState, cState, parent.SlotNumber()); !valid {
+				log.Debugf("Block failed validity conditions: %v", err)
```

Comment:
if a block fails validity, should it be debug or error? I think error is more appropriate here

Reply:
Debug - we'll see tons of blocks failing validity conditions from bad actors or bad proposers